File: review-service/src/main/java/com/shopflow/review/service/ReviewService.java
package com.shopflow.review.service;

import com.shopflow.review.dto.ReviewRequest;
import com.shopflow.review.dto.ReviewResponse;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.UUID;

public interface ReviewService {

    ReviewResponse submitReview(UUID userId, ReviewRequest request);

    Page<ReviewResponse> getProductReviews(UUID productId, int page, int size);

    ReviewResponse updateReview(UUID reviewId, UUID userId, ReviewRequest request);

    void deleteReview(UUID reviewId, UUID userId);

    void deleteReviewInternal(UUID reviewId);
}

File: shopflow-frontend/src/services/modules/review.service.js
import axiosInstance from '../api/axiosInstance';

export const reviewService = {
  submitReview: async (reviewData) => {
    return await axiosInstance.post('/reviews', reviewData);
  },

  getProductReviews: async (productId, page = 0, size = 10) => {
    const response = await axiosInstance.get(`/reviews/product/${productId}`, {
      params: { page, size }
    });
    return response.data;
  },

  updateReview: async (reviewId, reviewData) => {
    return await axiosInstance.put(`/reviews/${reviewId}`, reviewData);
  },

  deleteReview: async (reviewId) => {
    return await axiosInstance.delete(`/reviews/${reviewId}`);
  }
};
